README.md

```diff
@@ -5,21 +5,355 @@ Bazel CI
 # bazel-toolchains
 
 https://github.com/bazelbuild/bazel-toolchains is a repository where Google
-hosts Bazel toolchain configs. These configs are required to configure
-Bazel to issue commands that will execute inside a Docker container via a remote
-execution environment.
+hosts the source code for a CLI tool that can be used to generate Bazel toolchain configs. These
+configs are required to configure Bazel to issue commands that will execute inside a Docker
+container via a remote execution environment.
 
 These toolchain configs include:
 * C/C++ CROSSTOOL file,
 * BUILD file with toolchain rules, and
 * wrapper scripts.
 
-Release information of toolchain configs can be found at:
-https://releases.bazel.build/bazel-toolchains.html.
+# rbe_configs_gen - CLI Tool to Generate Configs
 
-This repository also hosts the skylark rule used to generate toolchain configs.
+[rbe_configs_gen](https://github.com/bazelbuild/bazel-toolchains/blob/master/cmd/rbe_configs_gen/rbe_configs_gen.go) is the
+a CLI tool written in Go that can be used to generate toolchain configs for a given combination
+of Bazel release and docker image. The output of the tool are toolchain configs in one or more of
+the following formats:
+* Tarball
+* Config files copied directly to a local directory
 
-This repo previously contained Bazel targets that are used to generate toolchain
-containers. Note that they have been migrated to the following two repos:
-* [layer-definitions](https://github.com/GoogleCloudPlatform/layer-definitions)
-* [container-definitions](https://github.com/GoogleCloudPlatform/container-definitions)
+rbe_configs_gen requires [docker](https://docs.docker.com/get-docker/) to be installed locally and
+internet access to work.
+
+Config users are recommended to use the CLI tool to generate and self host their own configs.
+Pre-generated configs will be provided for new releases of Bazel & the [RBE Ubuntu 16.04](https://console.cloud.google.com/marketplace/details/google/rbe-ubuntu16-04)
+without any SLOs. See [Pre-generated Configs](#pre-generated-configs) section below for details.
+
+The rest of this section describes how to use the rbe_configs_gen tool.
+
+## Building
+
+### Building using Docker on Linux (Recommended)
+
+Use the [official Golang docker image](https://hub.docker.com/_/golang) to build the rbe_configs_gen
+binary using Go 1.16. This avoids having to install the Go toolchain locally but requires
+[docker](https://docs.docker.com/get-docker/).
+
+1. Clone this repository and set it as the working directory:
+
+```bash
+$ git clone https://github.com/bazelbuild/bazel-toolchains.git
+$ cd bazel-toolchains
+```
+
+2. Run the following command:
+
+```bash
+$ docker run --rm -v $PWD:/srcdir -w /srcdir golang:1.16 go build -o rbe_configs_gen ./cmd/rbe_configs_gen/rbe_configs_gen.go
+```
+
+3. You should now be run `rbe_configs_gen` as follows:
```

Comment:
Run `rbe_configs_gen` as follows to see the flags it accepts:

Reply:
Or maybe provide the flags in this README? (maybe you don't want to do that in case the flags change and the README becomes stale).

Reply:
Done